src/routes/media.rs

```diff
@@ -67,11 +67,41 @@ fn process_file_upload(
     }
 }
 
+fn get_foreign_key(
+    key_name: &'static str,
+    entries: &Entries,
+) -> Result<Option<i32>, Custom<String>> {
+    let id_field = entries.fields.get(key_name).map(|field| &field[0].data);
+
+    // This isn't a map because we need to return out if parse fails
+    Ok(match id_field {
```

Comment:
I think this block should be a little less nested; it's kinda hard to reason about its behavior right now.

Reply:
Alright, sure